examples/data-objects/task-selection/src/oldestClientDiceRoller.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { OldestClientObserver } from "@fluid-experimental/task-manager";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IEvent } from "@fluidframework/common-definitions";
+import { assert } from "@fluidframework/common-utils";
+
+import { IDiceRoller } from "./interface";
+
+// The root is map-like, so we'll use this key for storing the value.
+const diceValueKey = "diceValue";
+
+/**
+ * The DiceRoller is our data object that implements the IDiceRoller interface.
+ */
+export class OldestClientDiceRoller extends DataObject implements IDiceRoller {
+    public static get ComponentName() { return "@fluid-example/oldest-client-dice-roller"; }
```

Comment:
Personally I'd prefer name to be part of factory, not object (consider case where we have multiple different factories that use same data object, but with slightly different semantics). I.e. code to refer to factory.type to get name

Reply:
Makes sense, will change.